src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -330,13 +321,12 @@ def create_canopy_index(self,
         # create index
         index_params = index_params or self._index_params
         try:
-            create_index(name=self.index_name,
-                         dimension=dimension,
-                         metadata_config={
-                             'indexed': indexed_fields
-                         },
-                         timeout=TIMEOUT_INDEX_CREATE,
-                         **index_params)
+            self._pinecone_client.create_index(
+                name=self.index_name,
+                dimension=dimension,
+                spec=spec,
+                timeout=TIMEOUT_INDEX_CREATE,
+                **index_params)
```

Comment:
Again, check me if I'm wrong but I think this is redundant given that all the params that use to go here are now part of `spec`

Reply:
metric and timeout are not part of spec, so worth keeping I guess